dev/provision_salt.sh

```diff
@@ -7,25 +7,34 @@
 set -e
 
 MINION_PLIST_PATH=/Library/LaunchDaemons/com.saltstack.salt.minion.plist
+LINUX_SALT_CLIENT_PATH="$HOME/salt-client"
+SALT_VERSION='3006.9'
```

Comment:
From https://github.com/flutter/flutter/issues/142627#issuecomment-1919922724, the x509 state started failing somewhere between versions 3006.1 and 3006.5.
3006.3 is known working and currently in use for at least some of the minions, so maybe that's a less volatile choice (at least for linux)?

Reply:
Done